Ubuntu/History/historymanager.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2021 UBports Foundation
+ *
+ * Authors:
+ *  Lionel Duboeuf <lduboeuf@ouvaton.org>
+ *
+ * This file is part of history-service.
+ *
+ * history-service is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * history-service is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "historymanager.h"
+#include "eventview.h"
+#include "manager.h"
+#include <QDebug>
+#include <QJSValueList>
+#include <QTimer>
+
+
+HistoryManager::HistoryManager(QObject *parent) :
+    QObject(parent), mPendingOperation(false)
+{
+}
+
+void HistoryManager::removeEvents(int eventType, const QString &from, const QJSValue &callback)
+{
+    if (!callback.isCallable()) {
+        qCritical() << "no callback found!";
+        return;
+    }
+
+    QJSValue result(callback);
+    if (mPendingOperation) {
+        result.call(QJSValueList { OperationError::OPERATION_ALREADY_PENDING });
```

Comment:
Generally one passes the error as the last parameter, so let's have ` { 0, 0, OperationError::OPERATION_ALREADY_PENDING }` here (and in the lines below) :-)